### PR TITLE
Update Rubocop config

### PR DESCRIPTION
**Pull request by ingytron**
Rubocop isn't working on master branch and it's because the config had changed but this repo is still configured to use the old config, resulting in a `404 "Not Found" while downloading remote config file ...` error. This PR updates it and makes rubocop runnable again.

Please note that this change generated a lot of `Correctable` offenses:
`61 files inspected, 1701 offenses detected, 1692 offenses auto-correctable`
For this, I added some custom cops to ignore a lot of the errors and then autocorrected the remaining 15. 

I'm definitely flexible when it comes to which cops to enable/disable but wanted to avoid changing too many files at once. 😄

### CODE REVIEW

File: lib/record_store/provider.rb [hunk 1]
```diff
@@ -152,24 +153,22 @@ def retry_on_connection_errors(
...
-            max_timeouts -= 1
-
-            $stderr.puts('Retrying after a connection timeout')
-          rescue Errno::ECONNRESET
-            raise if max_conn_resets <= 0
-            max_conn_resets -= 1
-
-            waiter.wait
-          end
+          return yield
```
Comment (ingytron) [inline lib/record_store/provider.rb:156]: Rubocop detected redundant `begin` statements, and that appears to be consistent with the [Ruby Style Guide](https://rubystyle.guide/#no-BEGIN-blocks)

Comment (ryandv) [reply lib/record_store/provider.rb:156]: This is a good cop to have on. :+1:

**Review by ryandv — APPROVED** [review]

**Review by XaF — APPROVED** [review]

File: lib/record_store.rb [hunk 2]
```diff
@@ -53,7 +53,7 @@ def secrets_path
     def zones_path
       @zones_path ||= Pathname.new(
         File.expand_path(config.fetch('zones_path'),
-        File.dirname(config_path)),
```
Comment (XaF) [inline lib/record_store.rb:56]: Not totally sure how I feel about that indentation, since we're still in the same parenthesis... 🤷

Comment (ingytron) [reply lib/record_store.rb:56]: I see what you mean but if I change the indentation there it detects as an alignment offense so I'd need to disable the `[Layout/ArgumentAlignment](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Layout/ArgumentAlignment)` cop to fix it. I suspect it's a Rubocop bug.

**Pull request merged**